app/scripts/controllers/network/provider-api-tests/helpers.js

```diff
@@ -0,0 +1,332 @@
+import nock from 'nock';
+import sinon from 'sinon';
+import { JsonRpcEngine } from 'json-rpc-engine';
+import { providerFromEngine } from 'eth-json-rpc-middleware';
+import EthQuery from 'eth-query';
+import createInfuraClient from '../createInfuraClient';
+
+/**
+ * @typedef {import('nock').Scope} NockScope
+ *
+ * A object returned by `nock(...)` for mocking requests to a particular base
+ * URL.
+ */
+
+/**
+ * @typedef {{makeRpcCall: (request: Partial<JsonRpcRequest>) => Promise<any>, makeRpcCallsInSeries: (requests: Partial<JsonRpcRequest>[]) => Promise<any>}} InfuraClient
+ *
+ * Provides methods to interact with the suite of middleware that
+ * `createInfuraClient` exposes.
+ */
+
+/**
+ * @typedef {{network: string}} WithInfuraClientOptions
+ *
+ * The options bag that `withInfuraClient` takes.
+ */
+
+/**
+ * @typedef {(client: InfuraClient) => Promise<any>} WithInfuraClientCallback
+ *
+ * The callback that `withInfuraClient` takes.
+ */
+
+/**
+ * @typedef {[WithInfuraClientOptions, WithInfuraClientCallback] | [WithInfuraClientCallback]} WithInfuraClientArgs
+ *
+ * The arguments to `withInfuraClient`.
+ */
+
+/**
+ * @typedef {{ nockScope: NockScope, blockNumber: string }} MockNextBlockTrackerRequestOptions
+ *
+ * The options to `mockNextBlockTrackerRequest`.
+ */
+
+/**
+ * @typedef {{ nockScope: NockScope, request: object, response: object, delay?: number }} MockSuccessfulInfuraRpcCallOptions
+ *
+ * The options to `mockSuccessfulInfuraRpcCall`.
+ */
+
+/**
+ * @typedef {{mockNextBlockTrackerRequest: (options: Omit<MockNextBlockTrackerRequestOptions, 'nockScope'>) => void, mockSuccessfulInfuraRpcCall: (options: Omit<MockSuccessfulInfuraRpcCallOptions, 'nockScope'>) => NockScope}} InfuraCommunications
```

Comment:
Is this type used anywhere?

Reply:
The next two appear to be unused as well

Reply:
It is being used — the names were just wrong. These have been corrected in [d9327ce](https://github.com/MetaMask/metamask-extension/pull/15556/commits/d9327ceed18649492056402b67276a4764c399d7).